src/traversals/maxadjvisit.jl

```diff
@@ -106,7 +106,7 @@ function _merge_vertex!(merged_vertices, fadjlist, is_merged, w, u, v)
         w[v2, root] = w[root, v2]
     end
     # update neighbors
-    fadjlist[root] = union(fadjlist[root], fadjlist[non_root])
+    union!(fadjlist[root], fadjlist[non_root])
```

Comment:
I believe linked lists would improve performance further (this needs to grow memory).
However, it seems they are broken atm (c.f. https://github.com/JuliaCollections/DataStructures.jl/issues/886)

Reply:
MutableLinkedLists are fixed by now, but they don't provide unique elements as Set does, so they are not suited here.